### PR TITLE
[SRE-34] backporting catalyst-multi-node.json dashboard to staging

### PR DESCRIPTION
**Pull request by pwilczynskiclearcode**

### CODE REVIEW

**Comment by hjpotter92** [comment]
iirc; they were duplicated by eli or me some time ago to test if the auto-backup workflow in this repository was working. we never got around to deleting and now i've forgotten which one was the backup :|

File: grafana/staging/dashboards/catalyst-multi-node-prod.json
Comment (pwilczynskiclearcode) [inline grafana/staging/dashboards/catalyst-multi-node-prod.json:1]: Removing `Catalyst Multi-Node (prod)` in staging. Let's have just a single dashboard.

File: grafana/staging/dashboards/catalyst-multi-node.json [hunk 1]
```diff
@@ -233,16 +354,30 @@
           "refId": "A"
         }
       ],
+      "thresholds": [
+        {
+          "colorMode": "critical",
+          "op": "gt",
+          "value": 300,
+          "visible": true
```
Comment (pwilczynskiclearcode) [inline grafana/staging/dashboards/catalyst-multi-node.json:362]: Not sure if threshold in staging is needed but doesn't cause any harm neither.

File: grafana/staging/dashboards/catalyst-multi-node-prod.json
Comment (pwilczynskiclearcode) [reply grafana/staging/dashboards/catalyst-multi-node-prod.json:1]:
```
diff grafana/staging/dashboards/catalyst-multi-node-prod.json grafana/staging/dashboards/catalyst-multi-node.json
75c75
<       "pluginVersion": "9.3.2",
---
>       "pluginVersion": "9.3.6",
83c83
<           "expr": "sum (mist_sessions{catalyst=\"true\", sessType=\"viewers\", stream!~\"$exclude\"})",
---
>           "expr": "sum(mist_sessions{catalyst=\"true\", sessType=\"viewers\", stream!~\"$exclude\"})",
136c136
<       "pluginVersion": "9.3.2",
---
>       "pluginVersion": "9.3.6",
144c144
<           "expr": "sum (mist_sessions{catalyst=\"true\", sessType=\"incoming\", stream!~\"$exclude\"})",
---
>           "expr": "sum(mist_sessions{catalyst=\"true\", sessType=\"incoming\", stream!~\"$exclude\"})",
197c197
<       "pluginVersion": "9.3.2",
---
>       "pluginVersion": "9.3.6",
348c348
<           "expr": "sum(mist_sessions{catalyst=\"true\", sessType=\"viewers\", stream!=\"000reallylongnonexistenstreamnamethatreallyshouldntexist000\", stream!~\"$exclude\"}[1m] ) by (stream) or on() vector(0)",
---
>           "expr": "sum(mist_sessions{catalyst=\"true\", sessType=\"viewers\", stream!=\"000reallylongnonexistenstreamnamethatreallyshouldntexist000\", stream!~\"$exclude\", stream!~\"catalyst_vod.+\"}[1m] ) by (stream) or on() vector(0)",
449c449
<           "expr": "sum (mist_sessions{catalyst=\"true\", sessType=\"incoming\", stream!~\"$exclude\"})",
---
>           "expr": "sum(mist_sessions{catalyst=\"true\", sessType=\"incoming\", stream!~\"$exclude\"})",
540c540
<           "expr": "sum (mist_sessions{catalyst=\"true\", sessType=\"viewers\", stream!~\"$exclude\"})",
---
>           "expr": "sum(mist_sessions{catalyst=\"true\", sessType=\"viewers\", stream!~\"$exclude\"})",
695c695
<           "expr": "sum (mist_sessions{catalyst=\"true\", sessType=\"viewers\", stream!~\"$exclude\", stream!~\"$exclude\"})",
---
>           "expr": "sum(mist_sessions{catalyst=\"true\", sessType=\"viewers\", stream!~\"$exclude\", stream!~\"$exclude\"})",
900c900,901
<           "expr": "sum(count_over_time({namespace=\"prod\", app=\"catalyst\"} |~ \"FAIL: onFail\" |~ \"Key duration mismatch; disconnecting\" | pattern \"<_> '<_>+<playback_id>': Key duration mismatch; disconnecting <ip> to recover\"[30m])) by (playback_id)",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({namespace=\"default\", app=\"catalyst\"} |~ \"FAIL: onFail\" |~ \"Key duration mismatch; disconnecting\" | pattern \"<_> '<_>+<playback_id>': Key duration mismatch; disconnecting <ip> to recover\"[30m])) by (playback_id)",
901a903
>           "queryType": "range",
953c955,956
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
1048c1051,1052
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
1088c1092,1093
<           "expr": "sum(count_over_time({app=\"catalyst\", namespace=\"prod\"} |~ \"Could not retrieve\" | pattern \"<_> https://<host>/\"[5m]))",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({app=\"catalyst\", namespace=\"default\"} |~ \"Could not retrieve\" | pattern \"<_> https://<host>/\"[5m]))",
1089a1095
>           "queryType": "range",
1097c1103,1104
<           "expr": "sum(count_over_time({app=\"ingress-nginx\", namespace=\"prod\", stream=\"stderr\"} != \"sent data after final chunk\" |~ \"[error]\" | pattern `<_> upstream: \"<protocol>://<ip>:<port>/<code>.<ext>\", host: \"<upstreamHost>:<upstreamPort>\"`  | port = \"4242\" | ext = \"json\"[1m])) ",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({app=\"ingress-nginx\", namespace=\"default\", stream=\"stderr\"} != \"sent data after final chunk\" |~ \"[error]\" | pattern `<_> upstream: \"<protocol>://<ip>:<port>/<code>.<ext>\", host: \"<upstreamHost>:<upstreamPort>\"` | port = \"4242\" | ext = \"json\"[1m])) ",
1099a1107
>           "queryType": "range",
1103c1111
<       "title": "\"Could not retrieve\" replications (prod)",
---
>       "title": "\"Could not retrieve\" replications",
1154c1162,1163
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
1253c1262,1263
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
1377c1387,1388
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
1463c1474
<         "message": "Catalyst node ${host} is unable to be replicated by lots of other servers. Please investigate this server and take it out of production if necessary.",
---
>         "message": "Catalyst node ${host} is unable to be replicated by lots of other servers. Please investigate this server and take it out of staging if necessary.",
1520c1531,1532
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
1561c1573,1574
<           "expr": "sum(count_over_time({app=\"catalyst\", namespace=\"prod\"} |~ \"Could not retrieve\" | pattern \"<_> https://<host>/\"[5m])) by (host)",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({app=\"catalyst\", namespace=\"default\"} |~ \"Could not retrieve\" | pattern \"<_> https://<host>/\"[5m])) by (host)",
1562a1576
>           "queryType": "range",
1570c1584,1585
<           "expr": "sum(count_over_time({app=\"ingress-nginx\", namespace=\"prod\", stream=\"stderr\"} != \"sent data after final chunk\" |~ \"[error]\" | pattern `<_> upstream: \"<protocol>://<ip>:<port>/<code>.<ext>\", host: \"<upstreamHost>:<upstreamPort>\"`  | port = \"4242\" | ext = \"json\"[1m])) by (upstreamHost)",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({app=\"ingress-nginx\", namespace=\"default\", stream=\"stderr\"} != \"sent data after final chunk\" |~ \"[error]\" | pattern `<_> upstream: \"<protocol>://<ip>:<port>/<code>.<ext>\", host: \"<upstreamHost>:<upstreamPort>\"` | port = \"4242\" | ext = \"json\"[1m])) by (upstreamHost)",
1572a1588
>           "queryType": "range",
1635c1651,1652
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
1671c1688,1690
<           "expr": "sum(count_over_time({namespace=\"prod\", app=\"ingress-nginx\", stream=\"stdout\"} != \"51eebae3-9b8e-48bc-89ca-5340c4074a48.json\" | pattern `<ip> - - <timestamp> \"<method> <url> <protocol>\" <status> <bytes> \"<originUrl>\" \"<userAgent>\" <wtf1> <wtf2> [<upstream>] [<wtf3>] <upstreamAddr> <wtf4> <wtf5> <upstreamStatus> <requestId>` | originUrl != \"-\" | line_format \"{{ .originUrl }}\" | pattern \"<originScheme>://<originDomain>/<originPath>\" | line_format \"{{ .originDomain }}\"[1m])) by (status)",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({namespace=\"default\", app=\"ingress-nginx\", stream=\"stdout\"} != \"51eebae3-9b8e-48bc-89ca-5340c4074a48.json\" | pattern `<ip> - - <timestamp> \"<method> <url> <protocol>\" <status> <bytes> \"<originUrl>\" \"<userAgent>\" <wtf1> <wtf2> [<upstream>] [<wtf3>] <upstreamAddr> <wtf4> <wtf5> <upstreamStatus> <requestId>` | originUrl != \"-\" | line_format \"{{ .originUrl }}\" | pattern \"<originScheme>://<originDomain>/<originPath>\" | line_format \"{{ .originDomain }}\"[1m])) by (status)",
>           "queryType": "range",
1688a1708,1709
>             "axisCenteredZero": false,
>             "axisColorMode": "text",
1721c1742,1743
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
1747c1769,1770
<           "mode": "single"
---
>           "mode": "single",
>           "sort": "none"
1757c1780,1781
<           "expr": "sum(count_over_time({namespace=\"prod\", app=\"catalyst\"} |~ \"Packet loss detected\" | pattern \"<_> MistOutTSSRT:<playback_id> \"[1m])) by (playback_id)",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({namespace=\"default\", app=\"catalyst\"} |~ \"Packet loss detected\" | pattern \"<_> MistOutTSSRT:<playback_id> \"[1m])) by (playback_id)",
1758a1783
>           "queryType": "range",
1766c1791,1792
<           "expr": "sum(count_over_time({namespace=\"prod\", app=\"catalyst\"} |~ \"Packet loss detected\" | pattern \"<_> MistOutTSSRT:<playback_id> \"[1m]))",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({namespace=\"default\", app=\"catalyst\"} |~ \"Packet loss detected\" | pattern \"<_> MistOutTSSRT:<playback_id> \"[1m]))",
1768a1795
>           "queryType": "range",
1785a1813,1814
>             "axisCenteredZero": false,
>             "axisColorMode": "text",
1818c1847,1848
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
1844c1874,1875
<           "mode": "single"
---
>           "mode": "single",
>           "sort": "none"
1853c1884,1885
<           "expr": "sum(count_over_time({app=\"catalyst\"} |~ \"not allowed\"[1m])) by (region)",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({app=\"catalyst\", namespace=\"default\"} |~ \"not allowed\"[1m])) by (region)",
1854a1887
>           "queryType": "range",
1871a1905,1906
>             "axisCenteredZero": false,
>             "axisColorMode": "text",
1904c1939,1940
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
1930c1966,1967
<           "mode": "single"
---
>           "mode": "single",
>           "sort": "none"
1939c1976,1977
<           "expr": "sum(count_over_time({app=\"catalyst\", namespace=\"prod\"} |~ \"Shutting down on Serf client failure\"[1m])) by (region, pod)",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({app=\"catalyst\", namespace=\"default\"} |~ \"Shutting down on Serf client failure\"[1m])) by (region, pod)",
1940a1979
>           "queryType": "range",
1957a1997,1998
>             "axisCenteredZero": false,
>             "axisColorMode": "text",
1990c2031,2032
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
2016c2058,2059
<           "mode": "single"
---
>           "mode": "single",
>           "sort": "none"
2025c2068,2069
<           "expr": "sum(count_over_time({app=\"catalyst\"} |~ \"Global configuration options not ready\"[1m])) by (region, pod)",
---
>           "editorMode": "code",
>           "expr": "sum(count_over_time({app=\"catalyst\", namespace=\"default\"} |~ \"Global configuration options not ready\"[1m])) by (region, pod)",
2026a2071
>           "queryType": "range",
2043a2089,2090
>             "axisCenteredZero": false,
>             "axisColorMode": "text",
2076c2123,2124
<                 "color": "green"
---
>                 "color": "green",
>                 "value": null
2102c2150,2151
<           "mode": "single"
---
>           "mode": "single",
>           "sort": "none"
2110a2160
>           "editorMode": "code",
2111a2162
>           "queryType": "range",
2153,2155c2204,2206
<   "title": "Catalyst Multi-Node (prod)",
<   "uid": "EWXkaj7Vz2",
<   "version": 2,
---
>   "title": "Catalyst Multi-Node",
>   "uid": "EWXkaj7Vz",
>   "version": 92,
```

File: grafana/staging/dashboards/catalyst-multi-node.json [hunk 2]
```diff
@@ -29,6 +29,10 @@
   "liveNow": false,
   "panels": [
     {
+      "datasource": {
+        "type": "prometheus",
+        "uid": "PBFA97CFB590B2093"
```
Comment (hjpotter92) [inline grafana/staging/dashboards/catalyst-multi-node.json:34]: wouldn't the uid be different for staging and prod prometheus data sources?

Comment (pwilczynskiclearcode) [reply grafana/staging/dashboards/catalyst-multi-node.json:34]: I thought so but looking at other staging files it's `PBFA97CFB590B2093` everywhere.

Comment (pwilczynskiclearcode) [reply grafana/staging/dashboards/catalyst-multi-node.json:34]: https://eu-metrics-monitoring.livepeer.monster/grafana/datasources

**Pull request merged**